Cilsil/Cil/Parsers/BinopParser.cs

```diff
@@ -34,7 +34,15 @@ internal class BinopParser : InstructionParser
                     break;
                 case Code.Cgt:
                 case Code.Cgt_Un:
-                    kind = BinopExpression.BinopKind.Gt;
+                    if (instruction.Previous.OpCode.Code == Code.Ldnull)
```

Comment:
Are you sure this is necessary? As far as I've known this has not caused an issue so far with regards to Infer's interpretation -- I believe it interprets obj > null as obj != null. So I would remove this modification.

Reply:
Not really. I saw obj > null caused null check unrecognized by Infer

Reply:
Just tested it -- you're right. However, the more precise handling would be to make the modification happen directly in PopTwoAndApplyBinop(...). It is not *essentially* important that the previous instruction as defined by Cecil is Ldnull. Rather, what is key is that you have a binop expression, the "kind" of the expression is BinopExpression.BinopKind.Gt, the rightExpression is a null int and the type is Tptr.

Reply:
Done. In this [commit](https://github.com/microsoft/infersharp/pull/57/commits/3d4687173eb44a74fb7c7cdbdbeb22f1a51b7078)